### ISSUE TITLE
Nowin app hangs using certificate for https after the first successful connection

### ISSUE THREAD

**Issue by dorvan** [Bobris/Nowin#55]
In a nowin app, with or without nancyfx (we use nancy for development, with and without razor)
using the certificate for https, the app hangs at first or max second  successfull connection, because cannot close connections to the certificate handshake, tcp ports on server side remain in "close wait" state, any new connection add another connection in this state without showing anyting, 
in our case this reproducible always in a linux 64 bit machine with mono 3.x > 4.3.x (git) 
with sources compiled in mono (framework 4.0) or in windows (4.2, 4.2.x) and running the compiled software in mono or official .net framwerk of both platforms, in debug mode in visual studio, the problem it's less frequently.

Looking other Bugs, seems quite similar to the "F5" bug (#52).
https://github.com/Bobris/Nowin/issues/52

sees a memory problem, or a certificate stream problem on the passthrough after the certicate loading and become a deadlock on tcp port communication after initial connection.
service LIsten correctly but handshake remain locked on the server side.

**Comment by Bobris** [comment]
I expect you are talking about latest version 0.22.0. So does it happen only in Mono or also in .Net FW on Windows?

**Comment by dorvan** [comment]
yes the latest, at moment we are back to owin due this problem, on linux with mono, the problem seems related to url binding and process initialization, the use of system.net it this way plus thread tasks cause instability.

**Comment by dorvan** [comment]
NO. It's not related, because we have tested all conbinations.
it's related on System.threading.tasks and System.Net calls. on Nowin.
